posts/2018-07-12-testing-in-the-wild.md

```diff
@@ -0,0 +1,277 @@
+---
+layout: post
+title: Testing in the wild
+
+meta:
+  nav: blog
+  author: etorreborre
+  pygments: true
+
+---
+
+ Writing tests seems like a wonderful idea in theory but real systems can be a real pain to test. Today I want to show a few tips on how to
+ use [specs2](http://specs2.org) + [ScalaCheck](http://www.scalacheck.org) to make some real-world testing somewhat bearable.
+
+ I am currently refactoring a big piece of code. Such refactoring is more like a small rewrite and some of our previous tests also have to be
+ rewritten from scratch. I will try to introduce you to the problem first.
+
+### Creating articles
+
+ The system-du-jour is called "Panda" (we have animal names for many of our services in my team) and is tasked with the creation of articles
+ on our legacy platform. An article is already a complicated beast. We have 3 levels of descriptions:
+
+  - Model: the description for a pair of Adidas shoes, with the brand, the gender it applies to, the size chart it uses and so on
```

Comment:
I would prefer no brand names in the post, lest it could be mistaken for advertisement.

Reply:
Sure will fix.